flang/lib/Lower/OpenMP/Decomposer.cpp

```diff
@@ -123,4 +123,9 @@ ConstructQueue buildConstructQueue(
 
   return constructs;
 }
+
+bool isLastItemInQueue(ConstructQueue::iterator item,
+                       const ConstructQueue &queue) {
+  return std::prev(queue.end()) == item;
```

Comment:
Comparing `std::next(item) == queue.end()` is more canonical.  Operations on the `end()` iterator may not work for all containers, but the "next" comparison will.